gunrock/app/geo/geo_test.cuh

```diff
@@ -0,0 +1,219 @@
+            SizeT degree  = graph.GetNeighborListLength(v);
+	
+	    SizeT i = 0;		
+
+	    // if location not known:
+	    if (!util::isValid(predicted_lat[v]) &&
+		!util::isValid(predicted_lon[v])) 
+	    {
+		// #pragma omp parallel
+	        for (SizeT k = 0; k < degree; k++) 
+		{
+		    SizeT e   = start_e + k;
+		    VertexT u = graph.GetEdgeDest(e);
+
+		    if (util::isValid(predicted_lat[u]) &&
+			util::isValid(predicted_lon[u])) 
+		    {
+			locations_lat[(v * degree) + i] = predicted_lat[u];
+			locations_lon[(v * degree) + i] = predicted_lon[u];
+			// printf("gather : < %lf , %lf >\n", predicted_lat[u], predicted_lon[u]);
+			i++;
+		    }
+
+		}
+		
+		valid_locations[v] = i;
+	    }
+        } // end: gather (for)
+  
+	// Compute operator 
+	// #pragma omp parallel
+	for (SizeT v = 0; v < nodes; ++v) 
+	{
+	    SizeT offset  = graph.GetNeighborListLength(v);
+	    if (!util::isValid(predicted_lat[v]) &&
+                !util::isValid(predicted_lon[v]))
+	    {
+
+//		util::PrintMsg("--- Spatial Center Operator --- ", !quiet);
+		spatial_center( locations_lat, 	      // list of neigh latitudes 
+				locations_lon, 	      // list of neigh longitudes
+				offset,		      // degree of vertex v	
+				valid_locations[v],   // number of valid locations
+				predicted_lat,	      // output latitude
+				predicted_lon,	      // output longitude
+				v,		      // active vertex
+			        quiet);	
+	    }
+	}
+
+	if (iterations >= geo_iter)
+	    Stop_Condition = true;
+
+	iterations++;
+
+	if(geo_complete) 
+	{	
+	    active = 0;
+
+	    // Check all nodes with known location,
+	    // and increment active.
+	    for (SizeT v = 0; v < nodes; ++v) 
+	    {
+	        if (util::isValid(predicted_lat[v]) && 
+	    	    util::isValid(predicted_lon[v])) 
+	        {
+		    active++;
+	        }
+	    }
+
+	    if(active == nodes) 
+	        Stop_Condition = true; 
+
+	    // util::PrintMsg("Current Predicted Locations: " 
+ 	    // 		+ std::to_string(active), !quiet);
+	}
+
+    } // -> while locations unknown.
+        // </TODO>
+
+    cpu_timer.Stop();
+    float elapsed = cpu_timer.ElapsedMillis();
+    return elapsed;
+}
+
+/**
+ * @brief Validation of geolocation results
+ * @tparam     GraphT        Type of the graph
+ * @tparam     ValueT        Type of the values
+ * @param[in]  parameters    Excution parameters
+ * @param[in]  graph         Input graph
+...
+ * @param[in]  verbose       Whether to output detail comparsions
+ * \return     GraphT::SizeT Number of errors
+ */
+template <typename GraphT>
+typename GraphT::SizeT Validate_Results(
+             util::Parameters &parameters,
+             GraphT           &graph,
+             typename GraphT::ValueT *h_predicted_lat,
+             typename GraphT::ValueT *h_predicted_lon,
+             typename GraphT::ValueT *ref_predicted_lat,
+             typename GraphT::ValueT *ref_predicted_lon,
+             bool verbose = true)
+{
```

Comment:
No actual correctness checking is being done here.  Presumably you'll need to check within some epsilon difference, since spatial median might converge to slightly different solutions depending on eg. order of the data